src/libraries/System.Private.DataContractSerialization/src/System/Runtime/Serialization/ClassDataContract.cs

```diff
@@ -704,6 +704,8 @@ internal bool RequiresMemberAccessForWrite(SecurityException? securityException)
         private sealed class ClassDataContractCriticalHelper : DataContract.DataContractCriticalHelper
         {
             private static Type[]? s_serInfoCtorArgs;
+            private static readonly MethodInfo s_getKeyValuePairMethod = typeof(KeyValuePairAdapter<,>).GetMethod("GetKeyValuePair", Globals.ScanAllMembers)!;
+            private static readonly ConstructorInfo s_ctorGenericMethod = typeof(KeyValuePairAdapter<,>).GetConstructor(Globals.ScanAllMembers, new Type[] { typeof(KeyValuePair<,>).MakeGenericType(typeof(KeyValuePairAdapter<,>).GetGenericArguments()) })!;
```

Comment:
@steveharter @GrabYourPitchforks @MichalStrehovsky - do any of you know an easier way to express this in reflection?

We are trying to find this constructor:

https://github.com/dotnet/runtime/blob/57bfe474518ab5b7cfe6bf7424a79ce3af9d6657/src/libraries/System.Private.DataContractSerialization/src/System/Runtime/Serialization/CollectionDataContract.cs#L26-L35

Reply:
I'm not immediately aware of any other way to write this that might play well with trimming. There are some tricks you can pull to save intermediate allocations, but since this is being placed into a readonly static, I don't know how much it would save in practice.

Reply:
There's no better thing that I know of. There was a proposal somewhere to complement [`Type.MakeGenericMethodParameter(Int32)`](https://docs.microsoft.com/en-us/dotnet/api/system.type.makegenericmethodparameter?view=net-5.0) with another API that would work for type generic parameters, but that didn't materialize.

Reply:
Thanks for the verification.